wagtail/admin/tests/pages/test_edit_page.py

```diff
@@ -1482,8 +1482,8 @@ def test_editor_page_shows_live_url_in_status_when_draft_edits_exist(self):
             reverse("wagtailadmin_pages:edit", args=(self.child_page.id,))
         )
 
-        input_field_for_draft_slug = '<input type="text" name="slug" value="revised-slug-in-draft-only" aria-describedby="panel-child-promote-child-for_search_engines-child-slug-helptext" id="id_slug" maxlength="255" required />'
-        input_field_for_live_slug = '<input type="text" name="slug" value="hello-world" aria-describedby="panel-child-promote-child-for_search_engines-child-slug-helptext" id="id_slug" maxlength="255" required />'
+        input_field_for_draft_slug = '<input type="text" name="slug" data-controller="w-slug" data-action="blur->w-slug#slugify" data-w-slug-allow-unicode-value value="revised-slug-in-draft-only" maxlength="255" aria-describedby="panel-child-promote-child-for_search_engines-child-slug-helptext" required="" id="id_slug">'
+        input_field_for_live_slug = '<input type="text" name="title" data-controller="w-slug" data-action="blur->w-slug#slugify" data-w-slug-allow-unicode-value value="Hello world!" maxlength="255" aria-describedby="panel-child-content-child-title-helptext" required id="id_title" />'
```

Comment:
I did some logging and it appears that the data attributes are not being added here, I think it is because the test panels are not using the ccustom Slug widget.

see `wagtail/test/testapp/models.py`

Reply:
@Lovelyfin00 I had another thought about this. Maybe the tests were failing for a different reason. It's probably best to leave the test app models as is.

However, this does open up the question about usage of Wagtail where they have added their own slug field for page and not used the common edit handlers. 

I think we should add our good friend - afterLoad to check for elements with id of id_slug and then convert them to a controlled element if they are not already. 

Another approach is to make the smart slug field the default for all slugs. Similar to how we do it for dates, this may be the best option as it simplifies the code changes and has less surprises for developers. 

They will need to opt out of the slug widget if they don't want it. See usage of `register_form_field_override` for how this is done. 

Any thoughts @thibaudcolas

Reply:
I have opted for the current approach for now, we can discuss at standup if we want to support the id_slug usage standalone - however this will require us to keep using the `window.allowUnicodeSlug` global into the next release which is not ideal.